nfc/labAttendancy.py

```diff
@@ -34,14 +38,53 @@ def on_connect(tag):
 
     sc = nfc.tag.tt3.ServiceCode(service_code >> 6, service_code & 0x3f)
     bc0 = nfc.tag.tt3.BlockCode(0, service=0)
+
+
+
     try:
         id_data = tag.read_without_encryption([sc], [bc0])
+    except Exception:
+        print("touch it again")
+        # print("showwarning",messagebox.showwarning("LabAttendancy","Touch it again!"))
+    else:
         id = id_data[2:13].decode("utf-8")
         print("Student Num: " + id)
-    except:
-        print("touch it again")
-
+        msg_info(id)
     return True
+def msg_info(id):
+# https://living-sun.com/ja/python-3x/683869-python-3-closing-window-on-tkinter-python-3x-tkinter.html
+    box = Tk()
+    box.title(u"Lab Attendency")
+    Label1 = Label(box,text = "Read Successful!",font=("Arial",30))
+    Label1.pack()
+    Label2 = Label(box,text = str(id),font=("Arial",20))
+    Label2.pack()
+    # 画面を中央に表示する
+    box.update_idletasks()
+    ww=box.winfo_screenwidth()
+    lw=box.winfo_width()
+    wh=box.winfo_screenheight()
+    lh=box.winfo_height()
+    box.geometry(str(lw)+"x"+str(lh)+"+"+str(int(ww/2-lw/2))+"+"+str(int(wh/2-lh/2)) )
+    box.after(1000, box.destroy) #秒数[ms]
+    box.mainloop()
+    return 0
+
+def msg_error():
+    box = Tk()
+    box.title(u"Lab Attendency")
+    Label1 = Label(box,text = "Touch It Again!",font=("Arial",30), fg="#ff0000",bg="#000000")
+    Label1.pack()
+    # 画面を中央に表示する
+    box.update_idletasks()
+    ww=box.winfo_screenwidth()
+    lw=box.winfo_width()
+    wh=box.winfo_screenheight()
+    lh=box.winfo_height()
+    box.geometry(str(lw)+"x"+str(lh)+"+"+str(int(ww/2-lw/2))+"+"+str(int(wh/2-lh/2)) )
+    box.after(1500, box.destroy) #秒数[ms]
+    box.mainloop()
+    return None
```

Comment:
ここもなんの関数なのかを明示したいところ
返り値も不要なので `return` だけでいいんじゃないかな

```suggestion

def msg_error():
    box = Tk()
    box.title(u"Lab Attendency")
    Label1 = Label(box, text = "Touch It Again!", font = ("Arial", 30), fg = "#ff0000", bg = "#000000")
    Label1.pack()
    box.update_idletasks()

    ww=box.winfo_screenwidth()
    lw=box.winfo_width()
    wh=box.winfo_screenheight()
    lh=box.winfo_height()
    box.geometry(str(lw) + "x" + str(lh) + "+" + str(int(ww/2-lw/2)) + "+" + str(int(wh/2-lh/2)))

    box.after(1500, box.destroy) #秒数[ms]
    box.mainloop()
    return
```